lib/web_ui/lib/src/engine/canvaskit/path_metrics.dart

```diff
@@ -20,9 +20,11 @@ class CkPathMetrics extends IterableBase<ui.PathMetric>
 
   /// The [CkPath.isEmpty] case is special-cased to avoid booting the WASM machinery just to find out there are no contours.
   @override
-  Iterator<ui.PathMetric> get iterator => _path.isEmpty
+  Iterator<ui.PathMetric> get iterator => _iterator ??= _path.isEmpty
       ? const CkPathMetricIteratorEmpty._()
       : CkContourMeasureIter(this);
+
+  Iterator<ui.PathMetric>? _iterator;
```

Comment:
Can this become a `late final`?
```suggestion
  late final Iterator<ui.PathMetric> iterator = _path.isEmpty
      ? const CkPathMetricIteratorEmpty._()
      : CkContourMeasureIter(this);
```

Reply:
done